tests/backup/backup_upgrade_test.go

```diff
@@ -204,3 +206,541 @@ var _ = Describe("{StorkUpgradeWithBackup}", func() {
+				err = CreateRestore(restoreName, backupName, namespaceMapping, destinationClusterName, orgID, ctx, storageClassMapping)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying restoration [%s] of backup [%s] in cluster [%s]", restoreName, backupName, restoreName))
+				restoreNames = append(restoreNames, restoreName)
+			}
+		})
+		// First singleNS schedule backups are taken before px-backup upgrade
+		Step("Restore first singleNS schedule backups", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			dash.VerifyFatal(err, nil, "Fetching px-central-admin ctx")
+			restoreSingleNSBackupInVariousWaysTask := func(index int, namespace string) {
+				firstSingleNSScheduleBackupName, err := GetFirstScheduleBackupName(ctx, singleNSScheduleNames[index], orgID)
+				log.FailOnError(err, "Getting first backup name of schedule [%s] failed", singleNSScheduleNames[index])
+				restoreConfigs := []struct {
+					namePrefix          string
+					namespaceMapping    map[string]string
+					storageClassMapping map[string]string
+					replacePolicy       ReplacePolicy_Type
+				}{
+					{
+						"test-restore-single-ns",
+						make(map[string]string, 0),
+						make(map[string]string, 0),
+						ReplacePolicy_Retain,
+					},
+					{
+						"test-custom-restore-single-ns",
+						map[string]string{namespace: "custom" + namespace},
+						make(map[string]string, 0),
+						ReplacePolicy_Retain,
+					},
+					{
+						"test-replace-restore-single-ns",
+						make(map[string]string, 0),
+						make(map[string]string, 0),
+						ReplacePolicy_Delete,
+					},
+				}
+				for _, config := range restoreConfigs {
+					restoreName := fmt.Sprintf("%s-%s", config.namePrefix, RandomString(4))
+					log.InfoD("Restoring first singleNS schedule backup [%s] in cluster [%s] with restore [%s] and namespace mapping %v", firstSingleNSScheduleBackupName, destinationClusterName, restoreName, config.namespaceMapping)
+					if config.replacePolicy == ReplacePolicy_Retain {
+						err = CreateRestore(restoreName, firstSingleNSScheduleBackupName, config.namespaceMapping, destinationClusterName, orgID, ctx, config.storageClassMapping)
+					} else if config.replacePolicy == ReplacePolicy_Delete {
+						err = CreateRestoreWithReplacePolicy(restoreName, firstSingleNSScheduleBackupName, config.namespaceMapping, destinationClusterName, orgID, ctx, config.storageClassMapping, config.replacePolicy)
+					}
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying restoration [%s] of first singleNS schedule backup [%s] in cluster [%s]", restoreName, firstSingleNSScheduleBackupName, restoreName))
+					restoreNames = SafeAppend(&mutex, restoreNames, restoreName).([]string)
+				}
+			}
+			_ = TaskHandler(singleNSNamespaces, restoreSingleNSBackupInVariousWaysTask, Parallel)
+		})
+		// By the time the next singleNS schedule backups are taken, the px-backup upgrade would have been completed
```

Comment:
nit: Better use `single namespace` instead of `singleNs`

Reply:
Changed all "singleNS" in string literals to "single namespace".

Reply:
Changed all "singleNS" in comments as well to "single namespace".